packages/mobile/src/verify/VerificationFailedModal.tsx

```diff
@@ -44,56 +44,88 @@ export function VerificationFailedModal(props: Props) {
+  const saltQuotaExceeded = props.verificationStatus === VerificationStatus.SaltQuotaExceeded
 
   const isVisible =
     (props.verificationStatus === VerificationStatus.Failed ||
       props.verificationStatus === VerificationStatus.RevealAttemptFailed ||
-      userBalanceInsufficient) &&
+      userBalanceInsufficient ||
+      saltQuotaExceeded) &&
     !isDismissed
 
   const allowEnterCodes = props.verificationStatus === VerificationStatus.RevealAttemptFailed
   // Only prompt forno switch if not already in forno mode and failure
   // wasn't due to insuffuicient balance
   const promptRetryWithForno =
-    props.retryWithForno &&
-    !props.fornoMode &&
-    props.verificationStatus !== VerificationStatus.InsufficientBalance
+    props.retryWithForno && !props.fornoMode && !userBalanceInsufficient && !saltQuotaExceeded
 
-  return promptRetryWithForno ? (
+  if (promptRetryWithForno) {
     // Retry verification with forno with option to skip verificaion
-    <WarningModal
-      isVisible={isVisible}
-      header={t('retryWithFornoModal.header')}
-      body1={t('retryWithFornoModal.body1')}
-      body2={t('retryWithFornoModal.body2')}
-      continueTitle={t('retryWithFornoModal.retryButton')}
-      cancelTitle={t('education.skip')}
-      onCancel={onSkip}
-      onContinue={onRetry}
-    />
-  ) : allowEnterCodes ? (
+    return (
+      <WarningModal
+        isVisible={isVisible}
+        header={t('retryWithFornoModal.header')}
+        body1={t('retryWithFornoModal.body1')}
+        body2={t('retryWithFornoModal.body2')}
+        continueTitle={t('retryWithFornoModal.retryButton')}
+        cancelTitle={t('education.skip')}
+        onCancel={onSkip}
+        onContinue={onRetry}
+      />
+    )
+  }
+
+  if (allowEnterCodes) {
     // Option to enter codes if reveal attempt failed
+    return (
+      <WarningModal
+        isVisible={isVisible}
+        header={t('failModal.header')}
+        body1={t('failModal.body1')}
+        body2={t('failModal.enterCodesBody')}
+        continueTitle={t('education.skip')}
+        cancelTitle={t('global:goBack')}
+        onCancel={onDismiss}
+        onContinue={onSkip}
+      />
+    )
+  }
+
+  if (userBalanceInsufficient) {
+    // Show userBalanceInsufficient message and skip verification
+    return (
+      <WarningModal
+        isVisible={isVisible}
+        header={t('failModal.header')}
+        body1={t('failModal.body1InsufficientBalance')}
+        body2={t('failModal.body2InsufficientBalance')}
+        continueTitle={t('education.skip')}
+        onContinue={onSkip}
+      />
+    )
+  }
+
+  if (saltQuotaExceeded) {
+    // Show saltQuotaExceeded message and skip verification
+    return (
+      <WarningModal
+        isVisible={isVisible}
+        header={t('failModal.header')}
+        body1={t('failModal.body1SaltQuotaExceeded')}
+        body2={t('failModal.body2SaltQuotaExceeded')}
+        continueTitle={t('education.skip')}
+        onContinue={onSkip}
```

Comment:
This is a recoverable error, we just need to send out a tx like the current Quota screen does. Is it intentional to skip here?

Reply:
I am wondering what the purpose of this modal is. My understanding is that when the quota is exceeded we automatically navigate to the quota purchase screen and this modal would only be visible if a users dismisses that screen. I think if we want to keep this modal, we'd want to eliminate the auto navigation and make it the CTA for this modal.

Reply:
@tarikbellamine you are right, this modal would only be visible if user has not funds to buy lookups or there was a failure during the purchase. But I think it makes sense to have, cause we want to explain why exactly user is not able to continue verification. Wdyt?

Reply:
I agree there is no harm in having this in but I think it may be a little out of place from a UX perspective.

The order right now is:
1. Try to verify but exceeded lookup quota
2. Get automatically routed to lookup purchase screen
3. Either dismiss the screen because you don't know what it is or encounter an error while purchasing, which you'd see in the error banner
4. Get routed back into verification flow and see the lookup quota error modal

It just feels weird to have step 4 end with a modal telling the user to purchase more lookups but we haven't given them any idea how to go do that. That's why I think switching step 2 and 4 could be good - we can use the error modal as an opportunity to inform them that they need to buy lookups to complete verification then the CTA on that model can navigate them to the lookup purchase screen.

Understand if it's a bit out of scope for this PR but what do you think?

Reply:
I think it make sense, but I would involve @nityas and @coreycelo in this. For this PR it's not critical, because it rather aims to fix broken flows. Great thinking on this one!